### DISCUSSION TITLE
Download Civitai Liked Models

### DISCUSSION THREAD

**Discussion by lundejd** [BlafKing/sd-civitai-browser-plus#92] [Ideas]
It would be great to have the ability to download all liked models from Civitai. My process is to browse and like anything I think would be interesting to use, but it is always a paint to go back in and download them all later. I'd love to be able to just have this extension handle it. This would also make it nice when moving between computers as you could set things up and quickly re-download everything without having to try copy them over, back them up etc.

**Comment by BlafKing** [comment]
To make the extension download all liked models I'd have to implement a download queue, which is something I'm currently looking into but I'm not sure if it'll be possible with the current implementation of the gradio progress bar.
(Or if it'll be possible at all with my python skills lol)

What should however definitely be possible is displaying/returning all the models you have liked.
To do this you'd have to create an API Key in your [account settings on CivitAI](https://civitai.com/user/account) and then input it in the CivitAI Browser settings tab.

I'll make an extra toggle appear if a personal API key is being used in the filter drop down that you can use to only show models that have been liked.

**Comment by lundejd** [comment]
That would do it, thank you!

From: Tom ***@***.***>
Date: Friday, November 3, 2023 at 2:16 PM
To: BlafKing/sd-civitai-browser-plus ***@***.***>
Cc: John Lundell ***@***.***>, Author ***@***.***>
Subject: Re: [BlafKing/sd-civitai-browser-plus] Download Civitai Liked Models (Discussion #92)

To make the extension download all liked models I'd have to implement a download queue, which is something I'm currently looking into but I'm not sure if it'll be possible with the current implementation of the gradio progress bar.

What should however definitely be possible is displaying/returning all the models you have liked.
To do this you'd have to create an API Key in your account settings on CivitAI<https://civitai.com/user/account> and then input it in the CivitAI Browser settings tab.

I'll make an extra toggle appear if a personal API key is being used in the filter drop down that you can use to only show models that have been liked.

**Comment by BlafKing** [comment]
I have now included a filter option which you can use to return only liked models!
I'll close this thread since the request is completed :)

(I'll also still look into the possibility of auto downloading all of them!)
